dotcom-rendering/src/article.ts

```diff
@@ -1,20 +1,41 @@
 import type { ArticleFormat } from '@guardian/libs';
-import { appsLightboxImages } from '../model/appsLightboxImages';
-import { buildLightboxImages } from '../model/buildLightboxImages';
-import { enhanceBlocks, enhanceMainMedia } from '../model/enhanceBlocks';
-import { enhanceCommercialProperties } from '../model/enhanceCommercialProperties';
-import { enhanceStandfirst } from '../model/enhanceStandfirst';
-import { enhanceTableOfContents } from '../model/enhanceTableOfContents';
-import { validateAsArticleType } from '../model/validate';
-import { type DCRArticle } from '../types/frontend';
-import { type RenderingTarget } from '../types/renderingTarget';
-import { decideFormat } from './decideFormat';
+import { decideFormat } from './lib/decideFormat';
+import type { ImageForAppsLightbox } from './model/appsLightboxImages';
+import { appsLightboxImages } from './model/appsLightboxImages';
+import { buildLightboxImages } from './model/buildLightboxImages';
+import { enhanceBlocks, enhanceMainMedia } from './model/enhanceBlocks';
+import { enhanceCommercialProperties } from './model/enhanceCommercialProperties';
+import { enhanceStandfirst } from './model/enhanceStandfirst';
+import { enhanceTableOfContents } from './model/enhanceTableOfContents';
+import { validateAsArticleType } from './model/validate';
+import type { ImageForLightbox } from './types/content';
+import type { FEArticleType } from './types/frontend';
+import { type RenderingTarget } from './types/renderingTarget';
+
+/**
+ * The `DCRArticle` type models the `FEArticleType` in addition to any enhancements DCR makes after
+ * receiving the data from Frontend.
+ */
+export type Article = FEArticleType & {
```

Comment:
All other 'main' types live in `src/types`. Could this change decrease discoverability by placing the types alongside the enhancers?

Reply:
Are you saying discoverability might decrease because this isn't in the `types` subdirectory, or because it combines the type and related functions in the same file, or both?

I'd argue two points:

1. Types and the core functions that create and operate on them should live together in the same file.
2. Subdirectories named using broad terms like "lib", "types" and "model" can have different meanings to different people, so it can be difficult to know where to put or find files in them. I'd argue for a flat structure, except in cases where we think it's particularly valuable to have a subdirectory and obvious what it's for, hence why I've put this file in `src`. In particular I don't think we should have a subdirectory called "types", as types are not a special category of things but exist everywhere in the codebase!

This is subjective and these are just arguments, so please let me know if you disagree, particularly if you think I'm missing something on the value of "lib", "types" and "model"!

Reply:
> Are you saying discoverability might decrease because this isn't in the types subdirectory, or because it combines the type and related functions in the same file, or both?

Both because the current convention is to have 'high level' types in `src/types`. I didn't create this convention but I assume our developers are expecting such types to live there.

I think co-location with related functions is more applicable if the type isn't used elsewhere. Given this is a fundamental type, used across the codebase and the current convention I think it might be less surprising for it to live in `src/types` along with all the other type definitions e.g. `FEArticleType`, `Palette`, the element types etc.

Not a blocker so I will leave it to you!